pennylane_lightning_gpu/src/tests/mpi/Test_MPI_StateVectorCudaManaged_NonParam.cpp

```diff
@@ -0,0 +1,2016 @@
+            init_state.size(), init_state.data(), indices.data(), false);
+
+        mpi_manager.Barrier();
+        sv.CopyGpuDataToHost(local_state.data(),
+                             static_cast<std::size_t>(subSvLength));
+        mpi_manager.Barrier();
+
+        CHECK(expected_local_state == Pennylane::approx(local_state));
+    }
+}
+
+TEMPLATE_TEST_CASE("StateVectorCudaMPI::SetIthStates",
+                   "[StateVectorCudaMPI_Nonparam]", float, double) {
+    using PrecisionT = TestType;
+    using cp_t = std::complex<PrecisionT>;
+    const std::size_t num_qubits = 5;
+
+    MPIManager mpi_manager(MPI_COMM_WORLD);
+
+    int nGlobalIndexBits =
+        std::bit_width(static_cast<unsigned int>(mpi_manager.getSize())) - 1;
+    int nLocalIndexBits = num_qubits - nGlobalIndexBits;
+    int subSvLength = 1 << nLocalIndexBits;
+    mpi_manager.Barrier();
+
+    int index;
+    if (mpi_manager.getRank() == 0) {
+        std::mt19937 re{1337};
+        std::uniform_int_distribution<> distr(
+            0, Pennylane::Util::exp2(num_qubits) - 1);
+        index = distr(re);
+    }
+    mpi_manager.Bcast(index, 0);
+
+    std::vector<cp_t> expected_state(Pennylane::Util::exp2(num_qubits), {0, 0});
+    if (mpi_manager.getRank() == 0) {
+        expected_state[index] = {1.0, 0};
+    }
+
+    auto expected_local_state = mpi_manager.scatter(expected_state, 0);
+    mpi_manager.Barrier();
+
+    int nDevices = 0; // Number of GPU devices per node
+    cudaGetDeviceCount(&nDevices);
+    int deviceId = mpi_manager.getRank() % nDevices;
+    cudaSetDevice(deviceId);
+
+    SECTION(
+        "Set Ith element of the state state on device with data on the host") {
+        StateVectorCudaMPI<PrecisionT> sv(mpi_manager, nLocalIndexBits,
+                                          nLocalIndexBits);
+        std::complex<PrecisionT> values = {1.0, 0};
+        sv.setBasisState(values, index, false);
+
+        std::vector<cp_t> h_sv0(subSvLength, {0.0, 0.0});
+        sv.CopyGpuDataToHost(h_sv0.data(),
+                             static_cast<std::size_t>(subSvLength));
+
+        CHECK(expected_local_state == Pennylane::approx(h_sv0));
+    }
+}
+
+TEMPLATE_TEST_CASE("StateVectorCudaMPI::applyPauliX",
+                   "[StateVectorCudaMPI_Nonparam]", float, double) {
+    using cp_t = std::complex<TestType>;
+    const std::size_t num_qubits = 4;
+
+    MPIManager mpi_manager(MPI_COMM_WORLD);
+
+    int nGlobalIndexBits =
+        std::bit_width(static_cast<unsigned int>(mpi_manager.getSize())) - 1;
+    int nLocalIndexBits = num_qubits - nGlobalIndexBits;
+    int subSvLength = 1 << nLocalIndexBits;
+    mpi_manager.Barrier();
+
+    std::vector<cp_t> init_sv = {
+        {0.0, 0.0}, {0.1, 0.0}, {0.2, 0.0}, {0.3, 0.0}, {0.4, 0.0}, {0.5, 0.0},
+        {0.6, 0.0}, {0.7, 0.0}, {0.8, 0.0}, {0.9, 0.0}, {1.0, 0.0}, {1.1, 0.0},
+        {1.2, 0.0}, {1.3, 0.0}, {1.4, 0.0}, {1.5, 0.0}};
+
+    std::vector<std::vector<cp_t>> expected_sv = {
```

Comment:
To avoid tracking these by hand, save some lines and maintenance, couldn't we just initialize a `StateVectorCudaManaged` object on each process, apply the gate, and compare its data with the `StateVectorCudaMPI` object?